applications/RANSApplication/custom_processes/rans_apply_exact_nodal_periodic_condition_process.cpp

```diff
@@ -0,0 +1,321 @@
+    Parameters default_parameters = Parameters(R"(
+        {
+            "base_model_part_name"           : "PLEASE_SPECIFY_MODEL_PART_NAME",
+            "master_model_part_name"         : "PLEASE_SPECIFY_MODEL_PART_NAME",
+            "slave_model_part_name"          : "PLEASE_SPECIFY_MODEL_PART_NAME",
+            "tolerance"                      : 1e-9,
+            "translation_settings" :
+            {
+                "direction" : [0.0, 0.0, 0.0],
+                "magnitude" : 0.0
+            },
+            "rotation_settings" :
+            {
+                "axis"   : [0.0, 0.0, 0.0],
+                "center" : [0.0, 0.0, 0.0],
+                "angle"  : 0.0
+            },
+            "echo_level"                     : 0,
+            "reorder"                        : true
+        })");
+
+    mrParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+    mBaseModelPartName = mrParameters["base_model_part_name"].GetString();
+    mMasterModelPartName = mrParameters["master_model_part_name"].GetString();
+    mSlaveModelPartName = mrParameters["slave_model_part_name"].GetString();
+    mReorder = mrParameters["reorder"].GetBool();
+    mTolerance = mrParameters["tolerance"].GetDouble();
+    mEchoLevel = mrParameters["echo_level"].GetInt();
+
+    const double eps = std::numeric_limits<double>::epsilon();
+
+    // translation settings
+    noalias(mTranslationDirection) =
+        mrParameters["translation_settings"]["direction"].GetVector();
+    const double translation_direction_norm = norm_2(mTranslationDirection);
+    if (translation_direction_norm > eps) {
+        noalias(mTranslationDirection) = mTranslationDirection / translation_direction_norm;
+    }
+    mTranslationMagnitude = mrParameters["translation_settings"]["magnitude"].GetDouble();
+
+    // rotation settings
+    noalias(mRotationCenter) = mrParameters["rotation_settings"]["center"].GetVector();
+    noalias(mRotationAxis) = mrParameters["rotation_settings"]["axis"].GetVector();
+    const double rotation_axis_norm = norm_2(mRotationAxis);
+    if (rotation_axis_norm > eps) {
+        noalias(mRotationAxis) = mRotationAxis / rotation_axis_norm;
+    }
+    mRotationAngle = mrParameters["rotation_settings"]["angle"].GetDouble();
+
+    KRATOS_ERROR_IF(mRotationAngle < eps && mTranslationMagnitude < eps)
+        << "Either translation or rotation should be defined.\n";
+    KRATOS_ERROR_IF(mRotationAngle > eps && rotation_axis_norm < eps)
+        << "Rotation angle defined, but rotation axis is not defined.\n";
+    KRATOS_ERROR_IF(mTranslationMagnitude > eps && translation_direction_norm < eps) << "Translation magnitude defined, but translation direction is not defined.\n";
+
+    const int domain_size =
+        mrModel.GetModelPart(mBaseModelPartName).GetProcessInfo()[DOMAIN_SIZE];
+    KRATOS_ERROR_IF(domain_size == 2 && (mRotationAxis[0] > eps || mRotationAxis[1] > eps))
+        << "2D rotation axis should be [0.0, 0.0, 1.0].\n";
```

Comment:
Why don't we automatically set this in the 2D case?

Reply:
done :)